### PR TITLE
fix: add update and create filter _cas field

### PR DESCRIPTION
**Pull request by superbogy**
fix: delete _type field in update action

issue:
#107 
@xavierchow

### CODE REVIEW

File: test/cb.test.js [hunk 1]
```diff
@@ -457,6 +457,29 @@ describe('Couchbase test', () => {
...
+          age: 24,
+          _cas: 'somethings'
+        };
+        await Person.destroyById(record.id);
+        await Person.create(record);
+        const connection = await Person.getConnector()._connection;
+        const { value } = await connection.getAsync(record.id);
+        value.should.have.property('name', record.name);
+        value.should.not.have.property('_cas');
+        await Person.update({ id: record.id }, Object.assign({ _type: 'test' }, record));
```
Comment (xavierchow) [inline test/cb.test.js]: why do you assign the `_type`?

File: lib/couchbaseX.js [hunk 2]
```diff
@@ -306,6 +306,9 @@ class CouchbaseAccessor extends Accessor {
    * Result is a promise with `[id, rev]` or an error.
    */
   postWithId(id, data, options) {
+    if ('_cas' in data) {
+      delete data._cas;
```
Comment (xavierchow) [inline lib/couchbaseX.js]: can you keep the original data unchanged? the caller might be surprised that `_cas` gets removed after the calling.

Comment (superbogy) [reply lib/couchbaseX.js]: modified

File: test/cb.test.js (hunk 1 above)
Comment (superbogy) [reply test/cb.test.js]: I want to test `_type ` was filtered in update

**Pull request merged**